include/SugarPHPMailer.php

```diff
@@ -407,4 +407,67 @@ protected function embeddedAttachmentExists($filename)
         return false;
     }
 
+    /**
+     * overloads PHPMailer::Send() to allow for better logging and debugging SMTP issues
+     *
+     * @return bool
+     */
+    public function send() 
+    {
+        //Use these to override some fields for tests when debugging SMTP issues:
+        //$this->Host     = "smtp.myserver.com";
+        //$this->From     = 'me@here.com';
+        //$this->FromName = 'My Name';
+        //$this->Sender   = 'me@here.com';
+        //$this->Password = 'wrong';
+        //$GLOBALS['log']->debug("PHPMailer Send Function: { FromName: $this->FromName From: $this->From Host: $this->Host UserName: $this->Username }");
+
+        $this->fullSmtpLog='';
+        $phpMailerExceptionMsg='';
+
+        try {
+            $saveExceptionsState = $this->exceptions;
+            $this->exceptions = true;
+
+            // pass callabck function for PHPMailer to call to provide log :
+            $this->Debugoutput = function($str, $level) {
+                // obfuscate part of response if previous line was a server 334 request, for authentication data: 
+                static $previousIs334 = false;  
+                if ($previousIs334) {
+                    $this->fullSmtpLog .= "$level: CLIENT -> SERVER: ---obfuscated---\n";
+                } else {
+                    $this->fullSmtpLog .= "$level: $str\n";
+                }
+                $previousIs334 = (strpos($str, 'SERVER -> CLIENT: 334') !== false);
+            };
+
+            $this->SMTPDebug = 3;
+            $ret = parent::send();
+            $this->exceptions =  $saveExceptionsState;
+        }
+        catch (Exception $e) {
+            $phpMailerExceptionMsg=$e->errorMessage(); //Pretty error messages from PHPMailer
+            if ($phpMailerExceptionMsg) {
+                $GLOBALS['log']->error("send: PHPMailer Exception: { $phpMailerExceptionMsg }");
+            }
+        }
+        catch (\Exception $e) { //The leading slash means the Global PHP Exception class will be caught
+            $phpMailerExceptionMsg=$e->getMessage(); //generic error messages from anything else
+            if ($phpMailerExceptionMsg) {
+                $GLOBALS['log']->error("send: PHPMailer Exception: { $phpMailerExceptionMsg }");
+            }
+        }
+
+        /* uncomment this to send full log into suitecrm.log:
+        // TODO: make this dependent on a flag set in config_override.php
```

Comment:
Please do not leave TODOs in code, implement it if needed or remove the commented section.